### PR TITLE
Added Google OAuth

### LINKED ISSUE #265: Add Google OAuth
### Description

This issue aims at creating a Google OAuth feature using google cloud and supabase. 
I am a Gssoc'24 contributor and would like to contribute. 

### Screenshots

_No response_

### Checklist

- [X] I have checked the existing [issues](https://github.com/pooranjoyb/popShop/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/pooranjoyb/popShop/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by masterboy376**
Added Google OAuth using google cloud and Supabase.
Admin is advised to configure Google OAuth on google cloud and Supabase after merging this PR.

Fixes #265

## Type of change

Please give a X on it which is applicable

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor Code
- [ ] A documentation update
- [ ] Others(mentioned in the issue number)

# Screenshots and Videos:

[screen-recorder-mon-jun-24-2024-18-18-33.webm](https://github.com/pooranjoyb/popShop/assets/92575005/a86bfbf4-cb68-4dfc-ab7c-3e26b4cb7d38)

# Checklist:
Give a X on, which is applicable

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by pooranjoyb — COMMENTED** [review]
Great job, @masterboy376! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

**Comment by masterboy376** [comment]
@pooranjoyb PTAL

**Comment by pooranjoyb** [comment]
> @pooranjoyb PTAL

show me the profile. how is the user created in `supabase`??? @masterboy376

**Comment by masterboy376** [comment]
@pooranjoyb, here is the screenshot
Screenshot (172)

**Comment by pooranjoyb** [comment]
yeah so you can see that we need to add something like `not-found`
Also are we saving this entire thing to Users table in supabase?

File: src/pages/Auth/Oauth.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+      if(data?.length === 0){
+        const { error } = await supabase.from("users").insert([
+          {
+            username: credential.user_metadata.email.split('@')[0],
+            email: credential.user_metadata.email,
+            password: hashedPassword,
+            firstname: credential.user_metadata.full_name.split(' ')[0],
+            lastname: credential.user_metadata.full_name.split(' ')[1],
+            gender: "",
```
Comment (pooranjoyb) [inline src/pages/Auth/Oauth.tsx:63]: update this

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

File: src/pages/Auth/Oauth.tsx (hunk 1 above)
Comment (masterboy376) [reply src/pages/Auth/Oauth.tsx]: done

**Comment by masterboy376** [comment]
@pooranjoyb PTAL 
Screenshot (173)

**Comment by masterboy376** [comment]
@pooranjoyb PTAL

**Comment by pooranjoyb** [comment]
> @pooranjoyb PTAL

Not working. Please check, something might be wrong :/

**Comment by masterboy376** [comment]
Hello @pooranjoyb 
This was happening due to domain mismatch in supabase redirect urls. Fixed it. Now login is working on both vercel deployments as well as local servers.
[screen-recorder-fri-jun-28-2024-07-50-58.webm](https://github.com/pooranjoyb/popShop/assets/92575005/176f3c0c-b073-4b68-8373-0ed19a5e2223)

**Comment by masterboy376** [comment]
@pooranjoyb  PTAL

**Comment by pooranjoyb** [comment]
> @pooranjoyb  PTAL

OK will review and update accordingly.

**Comment by pooranjoyb** [comment]
Hey @masterboy376 Do let me the environment variables I need to add for this feature. Otherwise everything looks good for the merge.

**Comment by masterboy376** [comment]
@pooranjoyb there is no additional env variable needs to be added

**Comment by masterboy376** [comment]
@pooranjoyb However I have created the Oauth consent screen using my personal google  cloud account. So, when ever you have time, do create a consent screen using organization's account.

**Review by pooranjoyb — APPROVED** [review]

**Comment by pooranjoyb** [comment]
> @pooranjoyb However I have created the Oauth consent screen using my personal google cloud account. So, when ever you have time, do create a consent screen using organization's account.

Yes, I do need to set this up. Drop me the steps below.

**Pull request merged**